src/openqaoa-core/openqaoa/backends/wrapper.py

```diff
@@ -49,6 +63,160 @@ def exact_solution(self, *args, **kwargs):
         return self.backend.exact_solution(*args, **kwargs)
 
 
+#------------------------ZNE WRAPPER---------------------------------
+available_factories = [
+    "Richardson",
+    "Linear",
+    "Poly",
+    "Exp",
+    "PolyExp",
+    "AdaExp",
+    "FakeNodes"
+]
+available_scaling = [
+    "fold_gates_at_random",
+    "fold_gates_from_right",
+    "fold_gates_from_left"
+]
+class ZNEWrapper(BaseWrapper):
+    """
+    This class inherits from the BaseWrapper and need to be backend agnostic
+    to QAOA implementations on different devices and their respectives SDKs. It 
+    implements Zero Noise Extrapolation (ZNE) from Mitiq framework. ZNE is an 
+    error mitigation technique used to extrapolate the noiseless expectation
+    value of an observable from a range of expectation values computed at 
+    different noise levels.
+
+    Parameters
+    ----------
+    factory: str
+        The name of the object that determines the zero-noise extrapolation method.
+    scaling: str
+        The name of the function for scaling the noise of a quantum circuit.
+    scale_factors: List[str]
+        Sequence of noise scale factors at which expectation values should be measured.
+        For AdaExpFactory, just the first element of the list will be considered.
+    order: int
+        Extrapolation order (degree of the polynomial fit). It cannot exceed len(scale_factors) - 1.
+        Just used for PolyFactory and PolyExpFactory.
+    steps: int
+        The number of optimization steps. At least 3 are necessary.
+        Just used for AdaExpFactory.
+    """
+
+    def __init__(self, backend, factory, scaling, scale_factors, order, steps):
+        super().__init__(backend)
+
+        # only qiskit backends are supported
+        assert( type(backend) == DEVICE_NAME_TO_OBJECT_MAPPER['qiskit.qasm_simulator'] or 
```

Comment:
Same comment here, maybe raise a `ValueError` instead of an `AssertionError`

Reply:
Addressed in the last commits